src/care/scan_impl.h

```diff
@@ -43,21 +44,49 @@ void exclusive_scan(chai::ManagedArray<T> data, //!< [in/out] Input data (output
    if (size > 0) {
       if (inPlace) {
          if (!data) {
-            printf("[CARE] Warning: Invalid arguments to care::exclusive_scan. If inPlace is true, data cannot be nullptr.");
+            printf("[CARE] Warning: Invalid arguments to care::exclusive_scan. If inPlace is true, data cannot be nullptr.\n");
             return;
          }
       }
       else {
          if (!outData) {
-            printf("[CARE] Warning: Invalid arguments to care::exclusive_scan. If inPlace is false, outData cannot be nullptr.");
+            printf("[CARE] Warning: Invalid arguments to care::exclusive_scan. If inPlace is false, outData cannot be nullptr.\n");
             return;
          }
          else if (size > 1 && !data) {
-            printf("[CARE] Warning: Invalid arguments to care::exclusive_scan. If inPlace is false and size > 1, data cannot be nullptr.");
+            printf("[CARE] Warning: Invalid arguments to care::exclusive_scan. If inPlace is false and size > 1, data cannot be nullptr.\n");
             return;
          }
       }
 
+#if !defined(CHAI_DISABLE_RM)
+      // Bounds checking is only available with the resource manager because the ManagedArray::size() is not
+      // reliable if it is cast to a different template type.
+      const chai::PointerRecord* dataRecord = chai::ArrayManager::getInstance()->getPointerRecord((void *)data.cdata());
```

Comment:
should I be explicitly including the umpire headers needed for the umpire calls?

Reply:
do you mean the chai calls? I could see an include of chai/PointerRecord.hpp (or whatever it is) to be explicit.

Reply:
Oops, I have to update the PR.... I meant the backtrace from umpire.